server/service/src/invoice/inbound_shipment/update/mod.rs

```diff
@@ -209,6 +216,70 @@ mod test {
         let context = service_provider.context().unwrap();
         let service = service_provider.invoice_service;
 
+        //InvoiceDoesNotExist
+        assert_eq!(
+            service.update_inbound_shipment(
+                &context,
+                &mock_store_a().id,
+                "n/a",
+                inline_init(|r: &mut UpdateInboundShipment| {
+                    r.id = "invalid".to_string();
+                    r.other_party_id = Some(mock_name_a().id.clone());
+                })
+            ),
+            Err(ServiceError::InvoiceDoesNotExist)
+        );
+        //NotAnInboundShipment
+        assert_eq!(
+            service.update_inbound_shipment(
+                &context,
+                &mock_store_a().id,
+                "n/a",
+                inline_init(|r: &mut UpdateInboundShipment| {
+                    r.id = mock_outbound_shipment_c().id.clone();
+                    r.other_party_id = Some(mock_name_a().id.clone());
+                })
+            ),
+            Err(ServiceError::NotAnInboundShipment)
+        );
+        //NotThisStoreInvoice
```

Comment:
Can still update invoice even if the store_id doesn't match 🤔

Reply:
I think create a new issue and add it in. not sure why it isn't being checked